packages/react-dom/src/__tests__/ReactFresh-test.internal.js

```diff
@@ -0,0 +1,1421 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @emails react-core
+ */
+
+/* eslint-disable no-for-of-loops/no-for-of-loops */
+
+'use strict';
+
+let React;
+let ReactDOM;
+let act;
+
+describe('ReactFresh', () => {
+  let container;
+  let familiesByID;
+  let familiesByType;
+  let newFamilies;
+  let updatedFamilies;
+  let performHotReload;
+  let signaturesByType;
+
+  beforeEach(() => {
+    let enableHotReloading;
+    let lastRoot;
+    global.__REACT_DEVTOOLS_GLOBAL_HOOK__ = {
+      supportsFiber: true,
+      inject: injected => {
+        enableHotReloading = injected.enableHotReloading;
+      },
+      onCommitFiberRoot: (id, root) => {
+        lastRoot = root;
+      },
+      onCommitFiberUnmount: () => {},
+    };
+
+    jest.resetModules();
+    React = require('react');
+    ReactDOM = require('react-dom');
+    act = require('react-dom/test-utils').act;
+    container = document.createElement('div');
+    document.body.appendChild(container);
+
+    familiesByID = new Map();
+    familiesByType = new WeakMap();
+
+    if (__DEV__) {
+      const {scheduleHotUpdate} = enableHotReloading(familiesByType);
```

Comment:
`scheduleHotUpdate` can be injected into DevTools the same as `overrideHookState` and `overrideProps`.

Reply:
Addressed.